core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -44,12 +41,11 @@ public FilteredQueryParser() {
     }
 
     @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
-        Query query = Queries.newMatchAllQuery();
-        Query filter = null;
-        boolean filterFound = false;
+        QueryBuilder query = new MatchAllQueryBuilder();
```

Comment:
if we end up using the match_all query builder twice as a default, maybe we should have a constant for this default in the builder?